packages/logic-utils/src/utils/xml-ns-key.ts

```diff
@@ -1,6 +1,5 @@
-//we are only allowing word (\w+) characters in prefixes now (in completions)
-//TODO it should be aligned with the full XML spec
-const namespaceRegex = /^xmlns(:(?<prefix>\w*))?$/;
+// The prefix is an xml name minus the ":"
+const namespaceRegex = /^xmlns(:(?<prefix>[^:=]*))?$/;
```

Comment:
by alignment I meant exporting the relevant `NAME` pattern from the XML Tokenizer.
- https://github.com/SAP/xml-tools/blob/master/packages/parser/lib/lexer.js#L35-L44

Your change is probably better so leave it, although instead of a subset it is now a superset so could even be incorrect in some cases.

But please keep the `TODO:` comment and maybe add the link:
> https://github.com/SAP/xml-tools/blob/master/packages/parser/lib/lexer.js#L35-L44

Reply:
I figured that we can rely on a lower level package to validate the attribute key and we can do a "best guess" that might contain invalid keys as long as it contains all the valid keys... so there it is.
Since the attribute key is already checked I don't think we have to re-validate it here.
I will add a comment that invalid attribute names are handled by the parser. We only validate the attribute value here and we can assume that if the user started the attribute key with `xmlns:` they meant it to be an xmlns attribute (even if it contains invalid characters).

Reply:
o.k. we can assume the prefix is a valid XML Identifier.
If it was not it would not be usable anyhow in a tag name.